src/Core_ExtensibleSaveFormat/Core.ExtendedSave.Extensions.cs

```diff
@@ -21,33 +21,36 @@ private static bool GetExtendedData(object messagePackObject, string id, out Plu
                 var bytes = (byte[])prop.GetValue();
                 if (bytes != null)
                 {
-                    Dictionary<string, PluginData> pluginData = MessagePackDeserialize<Dictionary<string, PluginData>>(bytes);
-
-                    if (pluginData != null && pluginData.TryGetValue(id, out data))
-                        return true;
+                    return MessagePackDeserialize<Dictionary<string, PluginData>>(bytes);
                 }
             }
             catch (System.Exception ex)
             {
                 Logger.LogError(ex);
             }
+            return null;
+        }
+
+        private static bool GetExtendedData(object messagePackObject, string id, out PluginData data)
+        {
+            Dictionary<string, PluginData> pluginData = GetAllExtendedData(messagePackObject);
+
+            if (pluginData != null && pluginData.TryGetValue(id, out data))
+                return true;
             data = null;
             return false;
         }
 
         private static void SetExtendedData(object messagePackObject, string id, PluginData data)
         {
+            Dictionary<string, PluginData> pluginData = GetAllExtendedData(messagePackObject);
+
+            if (pluginData == null) pluginData = new Dictionary<string, PluginData>();
+            pluginData[id] = data;
+            var bytes = MessagePackSerialize(pluginData);
```

Comment:
This should go inside the try block below, especially the serialize.

Reply:
sure